src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests/Requests/DataSetQueryCriteriaTimePeriodsValidatorTests.cs

```diff
@@ -0,0 +1,338 @@
+
+public class DataSetQueryCriteriaTimePeriodsValidatorTests
+{
+    private readonly DataSetQueryCriteriaTimePeriods.Validator _validator = new();
+
+    public static readonly TheoryData<DataSetQueryTimePeriod> ValidTimePeriodsSingle = new()
+    {
+        new DataSetQueryTimePeriod { Period = "2020", Code = "AY" },
+        new DataSetQueryTimePeriod { Period = "2020/2021", Code = "AY" },
+        new DataSetQueryTimePeriod { Period = "2020", Code = "FY" },
+        new DataSetQueryTimePeriod { Period = "2021", Code = "M1" },
+        new DataSetQueryTimePeriod { Period = "2021", Code = "W40" },
+        new DataSetQueryTimePeriod { Period = "2021", Code = "T3" },
+        new DataSetQueryTimePeriod { Period = "2021/2022", Code = "T3" },
+        new DataSetQueryTimePeriod { Period = "2019", Code = "FYQ4" },
+        new DataSetQueryTimePeriod { Period = "2019/2020", Code = "FYQ4" },
+    };
+
+    public static readonly TheoryData<DataSetQueryTimePeriod[]> ValidTimePeriodsMultiple = new()
+    {
+        new DataSetQueryTimePeriod[]
+        {
+            new() { Period = "2020", Code = "AY" },
+            new() { Period = "2020/2021", Code = "AY" },
+            new() { Period = "2020", Code = "FY" },
+        },
+        new DataSetQueryTimePeriod[]
+        {
+            new() { Period = "2021", Code = "M1" },
+            new() { Period = "2021", Code = "W40" },
+            new() { Period = "2021", Code = "T3" },
+        },
+        new DataSetQueryTimePeriod[]
+        {
+            new() { Period = "2021/2022", Code = "T3" },
+            new() { Period = "2019", Code = "FYQ4" },
+            new() { Period = "2019/2020", Code = "FYQ4" },
+        },
+    };
+
+    public static readonly TheoryData<DataSetQueryTimePeriod> InvalidTimePeriodsSingle = new()
+    {
+        new DataSetQueryTimePeriod { Period = "", Code = "AY" },
+        new DataSetQueryTimePeriod { Period = "Invalid", Code = "AY" },
+        new DataSetQueryTimePeriod { Period = "2022/2020", Code = "AY" },
+        new DataSetQueryTimePeriod { Period = "2000/1999", Code = "AY" },
+        new DataSetQueryTimePeriod { Period = "2022", Code = "" },
+        new DataSetQueryTimePeriod { Period = "2022/2023", Code = "" },
+        new DataSetQueryTimePeriod { Period = "2022", Code = "ay" },
+        new DataSetQueryTimePeriod { Period = "2022", Code = "Invalid" },
+        new DataSetQueryTimePeriod { Period = "2022", Code = "1" },
+        new DataSetQueryTimePeriod { Period = "2022", Code = "WEEK12" },
+        new DataSetQueryTimePeriod { Period = "2022/2023", Code = "JANUARY" },
+    };
+
+    public static readonly TheoryData<DataSetQueryTimePeriod[]> InvalidTimePeriodsMultiple = new()
+    {
+        Array.Empty<DataSetQueryTimePeriod>(),
+        new DataSetQueryTimePeriod[]
+        {
+            new() { Period = "", Code = "AY" },
+            new() { Period = "Invalid", Code = "AY" },
+            new() { Period = "2022/2020", Code = "AY" },
```

Comment:
Wonder if here you meant to put `2020/2022` to cover the case where the 2nd year isn't exactly **1** year after the 1st year. Because we already cover the 1st year being after the 2nd year in the line below

Reply:
Nope I deliberately wanted to test it being two years after, however, you're right, we should also test your case as well! I've added this in now.